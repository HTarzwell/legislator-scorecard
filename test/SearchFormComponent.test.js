import React from 'react'
import {
  SearchFormComponent
} from './../src/routes/landing/search/SearchFormComponent'
import { shallow, mount } from 'enzyme'
import toJSON from 'enzyme-to-json'

it('renders correctly', () => {
  const tree = shallow(<SearchFormComponent />)
  expect(toJSON(tree)).toMatchSnapshot()
})

function createNodeMock () {
  // You can return anything from this function.
  // For example:
  return {
    focus () {
      // Do nothing
    }
  }
}

it('allows you to submit the form', () => {
  const onFormSubmit = jest.spyOn(SearchFormComponent.prototype, 'onFormSubmit')
  const history = []
  const wrapper = mount(<SearchFormComponent history={history} />)

  wrapper.find('input[name="address"]').node.value = '123 Main St'
  wrapper.find('input[name="zipCode"]').node.value = '02139'

  wrapper.find('form').simulate('submit', {
    preventDefault: () => {}
  })
  expect(onFormSubmit).toHaveBeenCalled()
  expect(history[0]).toBe('/my-legislators/123 Main St MA 02139')
})
