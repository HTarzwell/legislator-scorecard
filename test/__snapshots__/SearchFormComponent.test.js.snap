// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="search-form my-5 my-md-0"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label>
      Address
      <input
        className="form-control"
        name="address"
        placeholder="123 Main St, Cambridge"
        type="text"
      />
    </label>
  </div>
  <div
    className="form-group"
  >
    <label>
      Zip Code
      <input
        className="form-control"
        name="zipCode"
        placeholder="02142"
        type="text"
      />
    </label>
  </div>
  <div
    className="form-group"
  >
    <label>
      State
      <input
        className="form-control"
        readOnly={true}
        type="text"
        value="Massachusetts"
      />
    </label>
  </div>
  <div
    className="mt-4"
  >
    <button
      className="btn btn-primary btn-block heading-font"
      type="submit"
    >
      Find My Local Legislators
    </button>
  </div>
</form>
`;
